sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoderSuite.scala

```diff
@@ -60,6 +61,22 @@ case class RepeatedData(
     mapFieldNull: scala.collection.Map[Int, java.lang.Long],
     structField: PrimitiveData)
 
+case class Nested(i: Option[Int], s: String)
+
+case class Data(
+    array: Seq[Int],
+    arrayContainsNull: Seq[Option[Int]],
+    map: scala.collection.Map[Int, Long],
+    mapContainsNul: scala.collection.Map[Int, Option[Long]],
+    nested: Nested)
+
+case class ComplexData(
+    arrayField: Seq[Int],
+    arrayFieldContainsNull: Seq[Option[Int]],
+    mapField: scala.collection.Map[Int, Long],
+    mapFieldContainsNull: scala.collection.Map[Int, Option[Long]],
+    dataField: Data)
+
```

Comment:
do we have to add these classes?
I think a simple test could be `encodeDecodeTest(Seq(Some(1), None), "seq of option")`